tests/dryrun_test.go

```diff
@@ -152,16 +151,9 @@ var _ = Describe("[sig-compute]Dry-Run requests", decorators.SigCompute, func()
 		var vm *v1.VirtualMachine
 		resource := "virtualmachines"
 
-		newVM := func() *v1.VirtualMachine {
-			vmiImage := cd.ContainerDiskFor(cd.ContainerDiskCirros)
-			vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(vmiImage, "echo Hi\n")
-			vm := libvmi.NewVirtualMachine(vmi)
-			return vm
-		}
-
 		BeforeEach(func() {
-			vm = newVM()
-
+			namespace := testsuite.GetTestNamespace(nil)
```

Comment:
I do not like this pattern of adding the namespace to the VM/VMI metadata.
I would have preferred that on creation, the namespace is specified and that is it. The duplication of this data has a potential to mess things up.

This pattern is repeated in this change all over. I do not know if it is intentional or just a continuation of the prev implementation.

Reply:
In general, you are right. but in this case, it failed several tests, where it created non-vm resources in vm.Namespace.